README.md

```diff
@@ -1,2 +1,152 @@
+* `semaphore.h`: for semaphores (ex. `sem_wait`, `sem_post`)
+* `sys/mman.h`: for memory mapping used in shared memory (ex. `mmap`)
+* `sys/time.h` and `time.h`: for measuring time (ex. `time_t`, `ctime`)
+* `Python.h`: for calling Python functions from C used in executor (ex. `Py_XDECREF`, `PyObject_CallObject`)
+* `signal.h`: for signal handlers and signal constants (ex. `signal`, `SIGINT`)
+* `string.h`: for string manipulation (ex. `strcpy`, `memset`)
+* `stdarg.h`: for variable-length arguments used in `logger` (ex. `va_args`)
+* `sys/stat.h`: for working with unconventional files (ex. `mkfifo`)
+* `sys/wait.h`: for wait functions used in executor to wait for processes (ex. `waitpid`)
+* `fcntl.h`: for constants used in opening and closing files (ex. `O_RDONLY`, `O_WRONLY`)
+
+## Running Runtime
+
+TODO!
+
+## Project Organization / Conventions
+
+We use Github Issues to track what we're doing and future initiatives to take up. These issues are loaded into a Github Project and assigned to someone. That person (or people) will create a branch off of whichever branch seems logical to branch off of, and do the work there. 
+
+Try to rebase your changes locally before pushing to Github, so that you don't clog up the public repo's history with small commits. 
+
+Be relatively descriptive in your commit messages (10 - 25 words) and prepend an all-caps tag enclosed in square brackets that describes the part of Runtime you were working on. For example:
+```
+[NET_HANDLER] logging to the network works but not thoroughly tested, deleted unused constants in net_util, added many comments to net handler files
+```
+When finished with the feature on your branch, submit a pull request to the branch that you branched off from and request the review of relevant people. Only after someone has reviewed your code and approved the pull request should you then merge (or rebase) your code onto your branch's parent branch (most often `master`).
+
+We do work in two-week cycles. At the beginning of each cycle (or "sprint"), everyone in the project sets goals for what they want to get done during that sprint. At the beginning of each meeting, everyone gives a short update on how they're doing, and what they intend to finish during that meeting. At the end of the cycle, each team member gives a demo of what they finished during that sprint, and the team gives feedback and evaluates the person's work. Of course, this requires the goals to be specific enough to demonstrate during a demo. For example, a goal of "I want to speed up the threads in `dev_handler`" is not specific enough. How will you demonstrate that during a demo? How will you quantify the speedup? How will you measure the speedup? In contrast, a goal of "The executor will be able to successfully terminate student code when Dawn disconnects" is specific enough. You can very clearly demonstrate that this occurs during a demo.
+
+TODO: For coding style conventions, see our style guide.
+
+## Useful Diagrams
```

Comment:
Maybe move this to before the explanation of each process so readers have a nice visual overview before going into the details.

Reply:
Done~~ I moved it to before `Overall Structure`